tests/cross_functional/system_test/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,214 @@
+            size=self.filesize, io_direction="write", storage_type="fs", end_fsync=1
+        )
+
+        self.pod_obj.get_fio_results()
+        logger.info(f"IO finished on pod {self.pod_obj.name}")
+
+    def flatten_image(self, clone_obj):
+        """
+        Flatten the image of clone
+
+        Args:
+            clone_obj: Object of clone of which image to be flatten
+        """
+        image_name = clone_obj.get_rbd_image_name
+        pool_name = "ocs-storagecluster-cephblockpool"
+
+        tool_pod = get_ceph_tools_pod()
+        out = tool_pod.exec_ceph_cmd(
+            ceph_cmd=f"rbd flatten {pool_name}/{image_name}",
+            format=None,
+        )
+        logger.info(f"{out}")
+        logger.info(f"Successfully flatten the image of {clone_obj.name}")
+
+    @skipif_external_mode
+    @magenta_squad
+    def test_clone_deletion_after_max_cluster_space_utilization(
+        self, interface_type, pvc_clone_factory, threading_lock
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
+            2. Create cephfs + RBD pvcs with data
+            3. Create clones of pvcs till cluster full ratio.
+            4. Make the cluster out of full by increasing the full ratio.
+            5. After the cluster is out of full state and IOs started , Try to delete clones.
+            6. Clone deletion should be successful and should not give error messages.
+        """
+        # Creating the clones one by one and wait until they bound
+        self.timeout = 1800
+        logger.info(
+            f"Start creating {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} GB."
+        )
+        clones_list = []
+
+        if interface_type == constants.CEPHBLOCKPOOL:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+
+                # flatten the image
+                self.flatten_image(cloned_pvc_obj)
+
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
+
+        else:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
```

Comment:
there is a redundancy in the code, the only diff I see is, in case of the rbd, you are flatting the clones.
Can you check if the below code does the same?

```for clone_num in range(self.num_of_clones + 1):
    logger.info(f"Start creation of clone number {clone_num}.")
    cloned_pvc_obj = pvc_clone_factory(
        self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
    )
    cloned_pvc_obj.reload()
    if interface_type == constants.CEPHBLOCKPOOL:
        # flatten the image
        self.flatten_image(cloned_pvc_obj)
    clones_list.append(cloned_pvc_obj)
    logger.info(
        f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
    )
```

Reply:
Yes. In case of rbd I am flattning the clones